faunadb-java-dsl/src/main/java/com/faunadb/client/types/Codec.java

```diff
@@ -24,12 +24,19 @@
  * <pre>{@code
  * class Person {
  *   static final Codec<Person> PERSON = new Codec<Person>() {
- *     public Result<Person> apply(Value value) {
+ *     public Result<Person> decode(Value value) {
  *       return Result.success(new Person(
- *         value.at("data", "firstName").to(Codec.STRING),
- *         value.at("data", "lastName").to(Codec.STRING)
+ *         value.at("firstName").to(Codec.STRING).getOrElse("<no name>"),
+ *         value.at("lastName").to(Codec.STRING).getOrElse("<no name>")
  *       ));
  *     }
+ *
+ *     public Result<Value> encode(Person person) {
+ *       return Result.success(new ObjectV(ImmutableMap.of(
```

Comment:
I'm sorry I didn't notice this yesterday. I think this example would be better if we leave the concrete values out of it. Something like:

```java
return Value.from(ImmutableMap.of(
  "firstName", person.firstName,
  "lastName", person.lastName
));
```